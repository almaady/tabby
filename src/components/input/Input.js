import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

/**
 * @class Input - It is an html input
 */
class Input extends React.PureComponent {
  /**
   * @property {string} name -The name of the input
   * @property {string} type - The type of the input
   * @property {string} placeholder- The placeholder of the input
   * @property {func} onBlur - The function triggered when the input is onBlur
   * @property {func} onChange - The function triggered when the input value change
   * @property {bool} showError - Determines if the input should add an error style
   * @property {string} defaultValue - The Default Value of the input
   * @property {func} innerRef - The function to create a Ref for the input
   * @property {string} autoComplete - To enable autocomplete
   * @property {bool} autoFocus - Makes the input focusable when is showed on the screen
   */
  static propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    showError: PropTypes.bool,
    defaultValue: PropTypes.string,
    innerRef: PropTypes.func,
    autoComplete: PropTypes.string,
    autoFocus: PropTypes.bool
  };

  render() {
    const {
      innerRef,
      name,
      type,
      placeholder,
      showError,
      onChange,
      onBlur,
      defaultValue,
      autoComplete,
      autoFocus
    } = this.props;
    return (
      <input
        ref={innerRef}
        autoComplete={autoComplete}
        name={name}
        type={type}
        placeholder={placeholder}
        className={classNames({ "input__error": showError }, "input")}
        onChange={onChange}
        onBlur={onBlur}
        defaultValue={defaultValue}
        autoFocus={autoFocus}
      />
    );
  }
}
Input.defaultProps = {
  defaultValue: "",
  placeholder: "",
  autoComplete: "off",
  showError: false,
  autoFocus: false,
  innerRef: () => {}

};

export default Input;
