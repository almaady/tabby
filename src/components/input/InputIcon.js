import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Icon from "../icons/Icon";
import Input from "./Input";

/**
 * @class Input - It is an html input
 */
class InputIcon extends React.PureComponent {
  /**
   * @property {string} name -The name of the input
   * @property {string} type - The type of the input
   * @property {string} placeholder- The placeholder of the input
   * @property {func} onBlur - The function triggered when the input is onBlur
   * @property {func} onChange - The function triggered when the input value change
   * @property {bool} showError - Determines if the input should add an error style
   * @property {string} defaultValue - The Default Value of the input
   * @property {func} innerRef - The function to create a Ref for the input
   * @property {string} autoComplete - To enable autocomplete
   * @property {bool} autoFocus - Makes the input focusable when is showed on the screen
   */
  static propTypes = {
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    onBlur: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    showError: PropTypes.bool,
    defaultValue: PropTypes.string,
    innerRef: PropTypes.func,
    autoComplete: PropTypes.string,
    icon: PropTypes.string.isRequired,
    iconColor: PropTypes.string,
    iconSize: PropTypes.string,
    autoFocus: PropTypes.bool
  };

  render() {
    const {
      innerRef,
      name,
      type,
      placeholder,
      showError,
      onChange,
      onBlur,
      defaultValue,
      autoComplete,
      icon,
      iconColor,
      iconSize,
      autoFocus
    } = this.props;
    return (
      <div className="input__icon--container">
        <Icon color={iconColor} icon={icon} size={iconSize}/>
        <Input 
          innerRef={innerRef}
          autoComplete={autoComplete}
          name={name}
          type={type}
          placeholder={placeholder}
          className={classNames({ "input__error": showError }, "input")}
          onChange={onChange}
          onBlur={onBlur}
          defaultValue={defaultValue}
          autoFocus={autoFocus}
        />
      </div>
    );
  }
}
InputIcon.defaultProps = {
  placeholder: "",
  defaultValue: "",
  innerRef: () => {},
  autoComplete: "off",
  iconColor: "gray-l2",
  iconSize: "R",
  showError: false,
  autoFocus: false
};

export default InputIcon;
