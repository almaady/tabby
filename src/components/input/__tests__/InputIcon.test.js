import React from "react"
import {shallow, mount} from "enzyme"
import sinon from "sinon"

import InputIcon from "../InputIcon"
import Input from "../Input";


describe("InputIcon", () => {
  const onBlur = sinon.stub();
  const onChange = sinon.stub();
  let props;
  beforeEach(() => {
    props = {
      name: "Input",
      type: "text",
      onBlur,
      onChange,
      icon: "down",
    };
  });


  it("should render correctly with default state", () => {
    const component = shallow(<InputIcon {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should add placeholder to the input", () => {
    const component = mount(<InputIcon {...props} placeholder="placeholder" />);
    expect(component.props().placeholder).toEqual("placeholder");
  });
  it("should call onChange when you change the input value", () => {
    const component = mount(<InputIcon {...props} />);
    component.find("input").simulate("change");
    expect(onChange.called).toBeTruthy();
  });
  it("should call onBlur when you focus the input ", () => {
    const component = mount(<InputIcon {...props} /> );
    component.find("input").simulate("blur");
    expect(onBlur.called).toBeTruthy();
  });
  it("should has an Icon ", () => {
    const component = mount(<InputIcon {...props} />)
    expect(component.find("Icon")).toBeTruthy();
  });
});
