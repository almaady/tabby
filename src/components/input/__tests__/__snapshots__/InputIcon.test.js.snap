// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputIcon should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputIcon
    autoComplete="off"
    autoFocus={false}
    defaultValue=""
    icon="down"
    iconColor="gray-l2"
    iconSize="R"
    innerRef={[Function]}
    name="Input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder=""
    showError={false}
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          color="gray-l2"
          icon="down"
          size="R"
        />,
        <Input
          autoComplete="off"
          autoFocus={false}
          className="input"
          defaultValue=""
          innerRef={[Function]}
          name="Input"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          showError={false}
          type="text"
        />,
      ],
      "className": "input__icon--container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "gray-l2",
          "icon": "down",
          "size": "R",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoComplete": "off",
          "autoFocus": false,
          "className": "input",
          "defaultValue": "",
          "innerRef": [Function],
          "name": "Input",
          "onBlur": [Function],
          "onChange": [Function],
          "placeholder": "",
          "showError": false,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            color="gray-l2"
            icon="down"
            size="R"
          />,
          <Input
            autoComplete="off"
            autoFocus={false}
            className="input"
            defaultValue=""
            innerRef={[Function]}
            name="Input"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            showError={false}
            type="text"
          />,
        ],
        "className": "input__icon--container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "gray-l2",
            "icon": "down",
            "size": "R",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoComplete": "off",
            "autoFocus": false,
            "className": "input",
            "defaultValue": "",
            "innerRef": [Function],
            "name": "Input",
            "onBlur": [Function],
            "onChange": [Function],
            "placeholder": "",
            "showError": false,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
